### PR TITLE
Add BaseRepository and some small syntax fixes

### PR DESCRIPTION
**Pull request by Sonickers**

### CODE REVIEW

File: lib/base_repository.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class BaseRepository
```
Comment (tobiaszwaszak) [inline lib/base_repository.rb:1]: it's not something that you have to change 
but you could also create `BaseConcept` in `app/concepts/base` this should be more clear for other developers every concepts related class are in `concepts` directory

**Review by tobiaszwaszak — APPROVED** [review]

**Review by wmaciejak — APPROVED** [review]
It is f*ckin awesome 🎉 
Like Tobiasz wrote - consider for the next step extraction BaseConcept class :)

**Review by KMosc — COMMENTED** [review]
How it works exactly? It seems cool, easy to read.

File: lib/base_repository.rb (hunk 1 above)
Comment (Sonickers) [reply lib/base_repository.rb:1]: I based my code on this:
https://github.com/monterail/mirror-consono-backend/blob/main/lib/base_repository.rb

Is that not considered best practices anymore? 🤔

Comment (wmaciejak) [reply lib/base_repository.rb:1]: You mean the directory? It depends how generally you treat `concepts` directory and there is no good answer.

- If `concepts` directory is considered as a directory separated for things separated in terms of business meaning, then I think that `base` shouldn't be here and in that way your current solution is good.
- If `concepts` directory is considered as a directory separated by our understanding of scopes in-app (cause this is also one of the approaches) then `base` might be here and inside might be the base repository.

It always depends on global understanding of `concepts` directory and there is always more than one good answer.

Comment (Sonickers) [reply lib/base_repository.rb:1]: Wow that's a lot of knowledge! 📚 🚀 Thank you!

**Pull request merged**